src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -137,7 +137,7 @@ cdef void plugin_get_metadata(
   cy_context.context = context
   self.plugin_callback(cy_context, python_callback)
 
-cdef void plugin_destroy_c_plugin_state(void *state):
+cdef void plugin_destroy_c_plugin_state(void *state) with gil:
```

Comment:
Huh. Arguably Cython's supposed to statically error out if such a thing is necessary and not present, so that's something to file on their tracker at some point, methinks...

But otherwise, yeah, works for me.